pylint/checkers/variables.py

```diff
@@ -2666,17 +2737,19 @@ def _check_unpacking(
         values = self._nodes_to_unpack(inferred)
         if values is not None:
             if len(targets) != len(values):
-                # Check if we have starred nodes.
-                if any(isinstance(target, nodes.Starred) for target in targets):
-                    return
+                if isinstance(inferred, DICT_TYPES):
+                    msg_id = "unbalanced-dict-unpacking"
+                else:
+                    msg_id = "unbalanced-tuple-unpacking"
                 self.add_message(
-                    "unbalanced-tuple-unpacking",
+                    msg_id,
                     node=node,
                     args=(
                         _get_unpacking_extra_info(node, inferred),
```

Comment:
This is a bit of a refactor but I would prefer if we could not put a `' '` at the beginning of this return value. As you can see in the generated documentation `*Possible unbalanced tuple unpacking with sequence%s:` looks very weird.